tools/topology/CMakeLists.txt

```diff
@@ -137,8 +137,8 @@ set(TPLGS
 	"sof-imx8qxp-nocodec-sai\;sof-imx8qxp-nocodec-sai"
 	"sof-imx8qxp-wm8960\;sof-imx8qxp-wm8960"
 	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682"
-	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682\;-DAMP_SSP=1"
-	"sof-tgl-max98373-rt5682\;sof-tgl-rt5682-ssp0-max98373-ssp2\;-DAMP_SSP=2"
+	"sof-tgl-max98373-rt5682-test\;sof-tgl-max98373-rt5682-test\;-DAMP_SSP=1"
+	"sof-tgl-max98373-rt5682-test\;sof-tgl-rt5682-ssp0-max98373-ssp2-test\;-DAMP_SSP=2"
```

Comment:
I really don't like all this. The kernel WILL NOT use those files as is, so they should be in a test directory. I would assume these topology rely on the Maxim mock-up component in the firmware, which IIRC is not part of the default build. If it is, then so we have so many topology files for the same hardware+firmware?

We really need to clarify what we test as a default (base firmware + base topologies) and stop assuming custom installs. This will be problematic for sof-bin releases if we continue this mix and match of custom firmware and custom topologies.

Reply:
> I really don't like all this. The kernel WILL NOT use those files as is, so they should be in a test directory. I would assume these topology rely on the Maxim mock-up component in the firmware, which IIRC is not part of the default build. If it is, then so we have so many topology files for the same hardware+firmware?
> 

Yes, this is for running the mock-up component on Maxim 98373 platforms, this could be somewhat 'test' feature for CI/developers only, but it *is* part of the default build for master/upstream, though chrome kconfig may override it.

I have no objection to put this to 'test' or 'development' or any other 'xyz' folders if that make things clearer.

> We really need to clarify what we test as a default (base firmware + base topologies) and stop assuming custom installs. This will be problematic for sof-bin releases if we continue this mix and match of custom firmware and custom topologies.

This is actually somewhat controversial here, we can't deliver sof-bin releases with 3rd party algorithms included (that's a difference with closed source FW), but the Linux machine driver is actually expected for this proprietary one, which make that all we are doing (including FW and tplg binaries) for this kind of machine just like some kinds of hacking or debugging feature only.

Reply:
> > I really don't like all this. The kernel WILL NOT use those files as is, so they should be in a test directory. I would assume these topology rely on the Maxim mock-up component in the firmware, which IIRC is not part of the default build. If it is, then so we have so many topology files for the same hardware+firmware?
> 
> Yes, this is for running the mock-up component on Maxim 98373 platforms, this could be somewhat 'test' feature for CI/developers only, but it _is_ part of the default build for master/upstream, though chrome kconfig may override it.

Why is it part of the master/upstream config in the first place?
This does not scale, we are not going to add a mockup for each and every algorithm in the default built.

@lgirdwood FYI, this seems rather odd...

> 
> I have no objection to put this to 'test' or 'development' or any other 'xyz' folders if that make things clearer.
> 
> > We really need to clarify what we test as a default (base firmware + base topologies) and stop assuming custom installs. This will be problematic for sof-bin releases if we continue this mix and match of custom firmware and custom topologies.
> 
> This is actually somewhat controversial here, we can't deliver sof-bin releases with 3rd party algorithms included (that's a difference with closed source FW), but the Linux machine driver is actually expected for this proprietary one, which make that all we are doing (including FW and tplg binaries) for this kind of machine just like some kinds of hacking or debugging feature only.

The machine driver should work with a baseline firmware and topology, with the test firmware and topology using the mockup algorithm, and with the actual firmware including the algorithm along with the topology referring to this proprietary component.

I don't see what dependencies we might have here. The machine driver is not going to instantiate any IPCs or deal with topology stuff.

Reply:
> > Yes, this is for running the mock-up component on Maxim 98373 platforms, this could be somewhat 'test' feature for CI/developers only, but it _is_ part of the default build for master/upstream, though chrome kconfig may override it.
> 
> Why is it part of the master/upstream config in the first place?
> This does not scale, we are not going to add a mockup for each and every algorithm in the default built.
> 
> @lgirdwood FYI, this seems rather odd...
> 
> > I have no objection to put this to 'test' or 'development' or any other 'xyz' folders if that make things clearer.
> > > We really need to clarify what we test as a default (base firmware + base topologies) and stop assuming custom installs. This will be problematic for sof-bin releases if we continue this mix and match of custom firmware and custom topologies.
> > 
> > 
> > This is actually somewhat controversial here, we can't deliver sof-bin releases with 3rd party algorithms included (that's a difference with closed source FW), but the Linux machine driver is actually expected for this proprietary one, which make that all we are doing (including FW and tplg binaries) for this kind of machine just like some kinds of hacking or debugging feature only.
> 
> The machine driver should work with a baseline firmware and topology, with the test firmware and topology using the mockup algorithm, and with the actual firmware including the algorithm along with the topology referring to this proprietary component.
> 
> I don't see what dependencies we might have here. The machine driver is not going to instantiate any IPCs or deal with topology stuff.

I think the test firmware with mockup algorithm is the baseline firmware and topology in the first place, that's why you are seeing odd thing here. E.g. we don't have any other baseline firmware and topology for the Maxim98373+rt5682 platform at the moment.